.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,259 @@
+        run: |
+          apt update -y
+          apt install -y gnupg2 curl
+      - name: Install Node.js
+        run: |
+          curl -fsSL https://deb.nodesource.com/setup_16.x | bash -
+          apt-get install -y nodejs
+      - name: Check Node.js version
+        run: node --version
+      - name: Verify files
+        run: |
+          curl -sSL https://secchannel.rsk.co/SUPPORT.asc | gpg2 --import -
+          gpg2 --verify SHA256SUMS.asc && sha256sum --check SHA256SUMS.asc
+
+      - uses: actions/cache@v4
+        name: Cache Gradle
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            rskj-core/build
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Build
+        run: |
+          ./configure.sh
+          ./gradlew --no-daemon dependencies
+          ./gradlew --no-daemon --stacktrace build -x test
+          ls -la ./gradle/wrapper/gradle-wrapper.jar
+
+  sonarqube:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Verify Java installation
+        run: java -version
+
+      - uses: actions/cache@v4
+        name: Restore Gradle cache
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            rskj-core/build
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get Gradle jar
+        run: |
+          ./configure.sh
+
+      - name: Run SonarQube analysis
+        run: |
+          sudo apt-get update && sudo apt-get install -yqq git
+          extra_flags=""
+          if [ -n "$CIRCLE_PULL_REQUEST" ]; then
+            # https://community.sonarsource.com/t/no-code-or-issue-found-in-pull-request-decorations-github-circleci/8496
+            git branch -f master origin/master
+            # extract PR number, as PR URLs are in the form
+            # https://github.com/$user/$repo/pull/$pr_number
+            pr_number==${{ github.event.pull_request.number }}
+            extra_flags="-Dsonar.pullrequest.base=master
+              -Dsonar.pullrequest.branch=${{ github.head_ref }}
+              -Dsonar.pullrequest.key=$pr_number"
+          else
+            extra_flags="-Dsonar.branch.name=master"
+          fi
+          pwd
+          ls -al
+          ./gradlew sonarqube --no-daemon -x build -x test \
+              $extra_flags \
+              -Dsonar.organization=rsksmart \
+              -Dsonar.host.url="https://sonarcloud.io" \
+              -Dsonar.login="${{ secrets.SONAR_TOKEN }}"
```

Comment:
didn't work with `-Dsonar.token=`. Can't remember why, but I did also find a stackoverflow comment suggesting to use `.login` for now.

Reply:
Weird. Sonar itself recommend using `sonar.token`. I'll take your word for it.

Reply:
let me try again now that it all works

Reply:
well must have been something else because this too now works :P